dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/harvest/client/HarvestingClientsServiceIT.java

```diff
@@ -0,0 +1,112 @@
+package edu.harvard.iq.dataverse.harvest.client;
+
+import edu.harvard.iq.dataverse.DataverseDao;
+import edu.harvard.iq.dataverse.DataverseSession;
+import edu.harvard.iq.dataverse.arquillian.arquillianexamples.WebappArquillianDeployment;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.persistence.harvest.HarvestingClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.transaction.api.annotation.TransactionMode;
+import org.jboss.arquillian.transaction.api.annotation.Transactional;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ejb.EJB;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Arquillian.class)
+@Transactional(TransactionMode.ROLLBACK)
+public class HarvestingClientsServiceIT extends WebappArquillianDeployment {
+    @PersistenceContext(unitName = "VDCNet-ejbPU")
+    private EntityManager em;
+
+    @Inject
+    private HarvestingClientsService harvestingClientsService;
+    @Inject
+    private HarvestingClientServiceBean harvestingClientService;
+    @Inject
+    private DataverseSession dataverseSession;
+    @EJB
+    private AuthenticationServiceBean authenticationServiceBean;
+    @Inject
+    private DataverseDao dataverseDao;
+
+    @Before
+    public void setUp() {
+        dataverseSession.setUser(authenticationServiceBean.getAdminUser());
+    }
+
+    @Test
+    public void shouldCreateHarvestingClient() {
+        // given
+        HarvestingClient newHarvestingClient = createHarvestingClient();
+
+        // when
+        harvestingClientsService.createHarvestingClient(newHarvestingClient);
+
+        // then
+        HarvestingClient dbHarvestingClient = harvestingClientService.findByNickname("newNickname");
+        assertNotNull(dbHarvestingClient);
+
+        assertEquals(dataverseDao.findByAlias("ownmetadatablocks").getId(), dbHarvestingClient.getDataverse().getId());
+        assertEquals("http://localhost:8080/oai", dbHarvestingClient.getHarvestingUrl());
+        assertEquals("newOaiSet", dbHarvestingClient.getHarvestingSet());
+        assertEquals("newMetadataFormat", dbHarvestingClient.getMetadataPrefix());
+        assertEquals("newHarvestingStyle", dbHarvestingClient.getHarvestStyle());
+        assertTrue(dbHarvestingClient.isScheduled());
+        assertEquals(HarvestingClient.SCHEDULE_PERIOD_WEEKLY, dbHarvestingClient.getSchedulePeriod());
+        assertEquals(7, dbHarvestingClient.getScheduleDayOfWeek().intValue());
+        assertEquals(12, dbHarvestingClient.getScheduleHourOfDay().intValue());
+    }
+
+    @Test
+    public void shouldUpdateHarvestingClient() {
```

Comment:
I wonder if we should keep the test, since it doesn't test anything (if you remove the service call it will also pass). But I guess it tests that service doesn't brake anything :P.

Reply:
Yeah. Dunno if there's a way to 'disconnect' managed bean after persisting it in db.

Reply:
You would have to manually manage transactions.

Reply:
I'd keep it the way it is then. There's no point in my opinion to over do  that test.